pkg/notification/api/validation.go

```diff
@@ -75,6 +95,26 @@ func (s *NotificationService) validateCreateSubscriptionNoCommandRequest(
 		}
 		return dt.Err()
 	}
+	// We should only save the tags if the Feature source type is in the list
+	if req.FeatureFlagTags != nil {
```

Comment:
Because the `feature_flags_tags` belongs to the `Feature` notification policy, we must validate it.